packages/activity/src/index.ts

```diff
@@ -255,7 +275,11 @@ export class Context {
     heartbeat: (details?: any) => void,
     logger: Logger
   ) {
-    this.info = info;
+    // Note that we don't freeze the context object itself, because some users add properties to it (generally from
+    // interceptors). We even had samples promoting that practice. We should probably deprecate that at some point, but
+    // lets give us some time before we do.
+    this.info = deepFreeze(info);
```

Comment:
I think it's fine if the context remains mutable and we allow users to attach to it, I'd even not freeze the info object, it's fine if we just mark it readonly. I haven't seen issues with this in over 2 years that we've had this SDK.